devito/passes/clusters/cse.py

```diff
@@ -117,19 +119,16 @@ def _compact_temporaries(exprs):
     # safely be compacted; a generic Symbol could instead be accessed in a subsequent
     # Cluster, for example: `for (i = ...) { a = b; for (j = a ...) ...`
     mapper = {e.lhs: e.rhs for e in exprs
-              if isinstance(e.lhs, Temp) and (q_leaf(e.rhs) or e.rhs.is_Function)}
+              if isinstance(e.lhs, Temp) and
+              (q_leaf(e.rhs) or isinstance(e.rhs, AbstractFunction))}
```

Comment:
something odd here.

`e.rhs` should be an `Indexed`, so it will never be an `AbstractionFunction`. Either you meant `e.rhs.function` or... well, it might be we can just drop that?